core/src/main/java/com/linecorp/armeria/common/logging/AbstractLogFormatterBuilder.java

```diff
@@ -206,4 +268,39 @@ public AbstractLogFormatterBuilder<T> contentSanitizer(
         responseContentSanitizer(contentSanitizer);
         return this;
     }
+
+    /**
+     * Checks whether the sanitizer and mask headers are not used together.
+     */
+    protected void validateSanitizerAndMaskHeaders() {
+        if (requestHeadersSanitizer() != null) {
+            if (maskHeaders() != null || maskRequestHeaders() != null) {
+                throw new IllegalArgumentException("requestHeaderSanitizer should not be used when " +
```

Comment:
I'm curious if we can just return a factory method so that we don't have to expose two methods which basically function similarly


e.g.
```java
LogFormatter.builderForJson()
            .headersSanitizer(LogFormatterSanitizers.headers("Cookie"))
            ...
            .headersSanitizer(LogFormatterSanitizers.builder()
                                                    .requestHeaders("Cookie")
                                                    .responseHeaders("Auth")
                                                    .build())
```

Do you have any thoughts @seonwoo960000 @ikhoon ?

Reply:
Seems great 👍 . Let's see if we are on the same page: 
- We can use `LogFormatterSanitizer` to configure masking options for `headers`, `requestHeaders` and `responseHeaders` 
- We still need to validate whether only one is configured between the `LogFormatterSanitizer` and `headersSanitizer/requestHeadersSanitizer/responseHeadersSanitizer`

Reply:
Oops, I didn't realize `LogFormatter` already has request/response sanitizers separately.

In such case, I guess the `LogFormatterSanitizers` doesn't need to have separate methods for `headers`, `requestHeaders` and `responseHeaders`:

```java
LogFormatter.builderForJson()
            .responseHeadersSanitizer(LogFormatterSanitizers.mask("Cookie"))
            .requestHeadersSanitizer(LogFormatterSanitizers.mask("Authorization"))
```
> We still need to validate whether only one is configured between the LogFormatterSanitizer and headersSanitizer/requestHeadersSanitizer/responseHeadersSanitizer

I think the current behavior of overriding is fine, but let me know if you feel differently

Reply:
It is a good idea to expose `LogFormatterSanitizers` as a utility API. Some users may want to customize the masking such as showing the first 4 letters.
`LogFormatterSanitizers` looks too general. I propose changing the name and adding factory methods to both support JSON and TEXT formats.
```java
LogFormatter.builderForJson()
            .responseHeadersSanitizer(HeadersSanitizers.ofJson().mask("Cookie"))
            .maskResponseHeaders("Cookie") // Shortcut
            .requestHeadersSanitizer(HeadersSanitizers.ofJson().mask("Authorization"))
            .maskRequestHeaders("Authorization") // Shortcut

LogFormatter.builder()
            .responseHeadersSanitizer(HeadersSanitizers.ofText().mask("Cookie"))
            .maskResponseHeaders("Cookie") // Shortcut
            .requestHeadersSanitizer(HeadersSanitizers.ofText().mask("Authorization"))
            .maskRequestHeaders("Authorization") // Shortcut
```

Reply:
@ikhoon, to wrap up, are the features we need are as follows? 
-  `HeadersSanitizer`, both for text(`builderForText()`) and json(`builderForJson()`). They will be used as an argument for `requestHeadersSanitizer`, `responseHeadersSanitizer`. 
- Shortcut methods such as `maskResponseHeaders` and `maskRequestHeaders`.

Reply:
Correct. We need different `HeadersSanitizers` for JSON and text to avoid creating an additional headers. `HeadersSanitizers` directly produces a serialized output for the target `LogFormatter`.

Reply:
@ikhoon, I've implemented as follows: 
- Added `HeadersSanitizers` and its implementations(`TextHeadersSanitizer`, `JsonHeadersSanitizer`) 
- `TextLogFormatter` and `JsonLogFormatter` are updated to use appropriate `HeadersSanitizer` implementations to sanitize request and response headers. 
- Sanitizing logic is performed by using both `requestHeadersSanitizer` and `maskRequestHeaders`(or `responseHeadersSanitizer` and `maskResponseHeaders`) 

---
```java
LogFormatter.builderForJson() 
                      .responseHeadersSanitizer(HeadersSanitizers.builderForJson().mask("Cookie")..build())
                      ...
```
In regard to above code, I haven't added `requestHeadersSanitizer` and `responseHeadersSanitizer` that receive `HeadersSanitizer` as an argument. The reasoning is as follows: 
- Is it useful for the users to define `HeadersSanitizer` for themselves and pass it as an argument ? Wouldn't it be sufficient to just have "as-is" `requestHeadersSanitizer` and new `maskRequestHeaders` method? Please correct me if I'm missing something 😄

Reply:
@minwoox Ok, I'll implement so that the API only includes `responseHeadersSanitizer` and `requestHeadersSanitizer`. Btw, having shortcut(`maskRequestHeaders`) was confusing me as well 😅.